feed/tests/test_feed_views.py

```diff
@@ -159,47 +165,32 @@ def test_like_view_return_warning_when_logged_out(self, post, client):
     def test_like_view_redirects_to_login_when_logged_out(self, post, client):
         # Testing that when logged out, if trying to access the like url
         # the user gets redirected to the login url
-        post_to_like = Post.posts.filter(title=post.title, content=post.content, author=post.author).first()
+        post_to_like = post
         response = client.get(post_like_url(post_to_like.id))
         assert response.status_code == REDIRECT_URL_STATUS
         assert response.url == LOGIN_URL
 
     def test_like_url_doesnt_adds_like_to_post(self, post, client):
         # Testing that when logged out, if trying to access the like url
         # it doesnt adds a like to a post
-        post_to_check_likes = Post.posts.filter(title=post.title, content=post.content, author=post.author).first()
-        assert post_to_check_likes.likes.all().count() == 0
+        post_to_check_likes = post
+        assert post_to_check_likes.likes.count() == 0
         client.get(post_like_url(post_to_check_likes.id))
-        assert post_to_check_likes.likes.all().count() == 0
+        assert post_to_check_likes.likes.count() == 0
 
-    def test_like_url_adds_like_to_post(self, post, client):
-        # Testing that the like url, adds a like to a specific post
-        post_to_like = Post.posts.filter(title=post.title, content=post.content, author=post.author).first()
-        like_user = authenticate(username=USERNAME1, password=USER_PASS)
 
-        if like_user is not None:
-            client.login(username=USERNAME1, password=USER_PASS)
-            assert like_user not in post_to_like.likes.all()
-            client.get(post_like_url(post_to_like.id))
-            assert like_user in post_to_like.likes.all()
-        else:
-            # Login details are invalid
-            assert False
+class TestLikeForLoggedUser:
+    def test_like_url_adds_like_to_post(self, post, logged_in_client):
+        # Testing that the like url, adds a like to a specific post
+        post_to_like = post
+        assert post_to_like.likes.count() == 0
```

Comment:
test verification should be done through client url, isnt it?

Reply:
Done